File: flowman-common/src/main/scala/com/dimajix/flowman/common/ConsoleColors.scala
/*
 * Copyright 2020-2022 Kaya Kupferschmidt
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.dimajix.flowman.common


object ConsoleColors {
    private var disabled = false
    def setColorEnabled(enabled:Boolean) : Unit = {
        disabled = !enabled
    }

    val NORMAL = Console.RESET
    val CYAN = Console.CYAN
    val CYAN_BOLD = Console.CYAN + Console.BOLD
    val YELLOW = Console.YELLOW
    val YELLOW_BOLD = Console.YELLOW + Console.BOLD
    val RED = Console.RED
    val RED_BOLD = Console.RED + Console.BOLD
    val GREEN = Console.GREEN
    val GREEN_BOLD = Console.GREEN + Console.BOLD
    val WHITE = Console.WHITE
    val WHITE_BOLD = Console.WHITE + Console.BOLD

    def white(str:String) : String = if (disabled) str else WHITE + str + NORMAL
    def boldWhite(str:String) : String = if (disabled) str else WHITE_BOLD + str + NORMAL
    def green(str:String) : String = if (disabled) str else GREEN + str + NORMAL
    def boldGreen(str:String) : String = if (disabled) str else GREEN_BOLD + str + NORMAL
    def red(str:String) : String = if (disabled) str else RED + str + NORMAL
    def boldRed(str:String) : String = if (disabled) str else RED_BOLD + str + NORMAL
    def yellow(str:String) : String = if (disabled) str else YELLOW + str + NORMAL
    def boldYellow(str:String) : String = if (disabled) str else YELLOW_BOLD + str + NORMAL
    def cyan(str:String) : String = if (disabled) str else CYAN + str + NORMAL
    def boldCyan(str:String) : String = if (disabled) str else CYAN_BOLD + str + NORMAL
}

File: flowman-common/src/main/java/com/dimajix/flowman/common/ConsoleColors.java
/*
 * Copyright 2020-2022 Kaya Kupferschmidt
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.dimajix.flowman.common;


import scala.Console;

public class ConsoleColors {
    private static boolean disabled = false;
    public static void setColorEnabled(boolean enabled) {
        disabled = !enabled;
    }

    public static final String NORMAL = Console.RESET();
    public static final String CYAN = Console.CYAN();
    public static final String CYAN_BOLD = Console.CYAN() + Console.BOLD();
    public static final String YELLOW = Console.YELLOW();
    public static final String YELLOW_BOLD = Console.YELLOW() + Console.BOLD();
    public static final String RED = Console.RED();
    public static final String RED_BOLD = Console.RED() + Console.BOLD();
    public static final String GREEN = Console.GREEN();
    public static final String GREEN_BOLD = Console.GREEN() + Console.BOLD();
    public static final String WHITE = Console.WHITE();
    public static final String WHITE_BOLD = Console.WHITE() + Console.BOLD();

    public static String white(String str)  { return disabled ? str : WHITE + str + NORMAL; }
    public static String boldWhite(String str)  { return disabled ? str : WHITE_BOLD + str + NORMAL; }
    public static String green(String str)  { return disabled ? str : GREEN + str + NORMAL; }
    public static String boldGreen(String str)  { return disabled ? str : GREEN_BOLD + str + NORMAL; }
    public static String red(String str)  { return disabled ? str : RED + str + NORMAL; }
    public static String boldRed(String str)  { return disabled ? str : RED_BOLD + str + NORMAL; }
    public static String yellow(String str)  { return disabled ? str : YELLOW + str + NORMAL; }
    public static String boldYellow(String str)  { return disabled ? str : YELLOW_BOLD + str + NORMAL; }
    public static String cyan(String str)  { return disabled ? str : CYAN + str + NORMAL; }
    public static String boldCyan(String str)  { return disabled ? str : CYAN_BOLD + str + NORMAL; }
}
